habitat_baselines/common/base_trainer.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+from habitat_baselines.common.trainer_registry import trainer_registry
+
+
+class BaseTrainer(ABC):
```

Comment:
To stay uniform with other places in the codebase, let's avoid inheriting from `ABC` and instead using, `raise NotImplementedError`. 

Note: The ABC design is definitely valid but it is something which the libraries from our domain don't use (eg: pytorch), hence `ABC` was not included in the original design.